tests/foreman/cli/test_leapp_client.py

```diff
@@ -105,3 +118,165 @@ def test_positive_leapp_upgrade_rhel(
+    module_target_sat.cli.ActivationKey.update(
+        {
+            'id': function_leapp_ak.id,
+            'organization-id': module_sca_manifest_org.id,
+            'release-version': RHEL8_VER,
+        }
+    )
+
+    # re-register host with pull provider capsule
+    result = custom_leapp_host.register(
+        module_sca_manifest_org,
+        smart_proxy_location,
+        function_leapp_ak.name,
+        module_capsule_configured_mqtt,
+        setup_remote_execution_pull=True,
+        repo_data=f'repo={client_repo.baseurl}',
+        ignore_subman_errors=True,
+        force=True,
+    )
+    assert result.status == 0, f'Failed to register host: {result.stderr}'
+
+    service_name = get_yggdrasil_service_name(custom_leapp_host)
+    result = custom_leapp_host.execute(f'systemctl status {service_name}')
+    assert result.status == 0, f'Failed to start yggdrasil on client: {result.stderr}'
+
+    # unset releasever in AK
+    module_target_sat.cli.ActivationKey.update(
+        {
+            'id': function_leapp_ak.id,
+            'organization-id': module_sca_manifest_org.id,
+            'release-version': '',
+        }
+    )
+
+    # Workaround for https://issues.redhat.com/browse/RHEL-55871
+    assert custom_leapp_host.execute('echo \'ulimit -n 16384\' > /root/.bashrc').status == 0
+    # Run leapp preupgrade job
+    invocation_command = module_target_sat.cli_factory.job_invocation_with_credentials(
+        {
+            'job-template': 'Run preupgrade via Leapp',
+            'search-query': f'name = {custom_leapp_host.hostname}',
+            'organization-id': org.id,
+        },
+        (login, password),
+    )
+    result = module_target_sat.cli.JobInvocation.info({'id': invocation_command['id']})
+    assert result['success'] == '1'
+
+    # Run leapp upgrade job
+    invocation_command = module_target_sat.cli_factory.job_invocation_with_credentials(
+        {
+            'job-template': 'Run upgrade via Leapp',
+            'search-query': f'name = {custom_leapp_host.hostname}',
+            'organization-id': org.id,
+            'inputs': 'Reboot=false',
+        },
+        (login, password),
+    )
+
+    custom_leapp_host.power_control(state='reboot')
+    result = module_target_sat.cli.JobInvocation.info({'id': invocation_command['id']})
+    assert result['success'] == '1'
+
+    custom_leapp_host.clean_cached_properties()
+    assert str(custom_leapp_host.os_version) == upgrade_path['target_version']
+
+    # check mqtt still works after upgrade
+    service_name = get_yggdrasil_service_name(custom_leapp_host)
+    result = custom_leapp_host.execute(f'systemctl status {service_name}')
+    assert result.status == 0, f'Failed to start yggdrasil on client: {result.stderr}'
+
+    invocation_command = module_target_sat.cli_factory.job_invocation_with_credentials(
+        {
+            'job-template': 'Run Command - Script Default',
+            'inputs': "command=ls",
```

Comment:
Test LGTM, I wonder why the REX is needed to run the ls command in the end?

Reply:
Its just an additional check to be sure, even if the service is up on the client, there could potentially be some problem with the actual execution